Add top-level save and render utilities (#3134)
ioam/holoviews: doc/FAQ.rst

  curve = hv.Curve(df, 'x_col', 'y_col')
  curve = curve.redim.range(x_col=(0, None))

> Much like labels this allows you to declaratively associate ranges

Reviewer: ```suggestion
As in the discussion of labels above, this approach allows you to declaratively associate ranges
```

with the dimensions of your data in a way that will persist even if
you apply operations to the object. This same method is applicable to
adjust the range of a color bar. Here z_col is the color bar value

> dimension and is bounded from 0 to 5.

Reviewer: Where is "Here"?  "Below"?

To override the range specifically for plotting it is also possible to
set the `xlim` and `ylim` plot options:

.. code:: python

>   hv.Curve(df, 'x_col', 'y_col').options(xlim=(0, None), ylim=(0, 10))

Reviewer: ```suggestion
  hv.Curve(df, 'x_col', 'y_col').options(xlim=(0, None), ylim=(0, 10))
  
This approach allows you to customize objects easily as a final step, but note that the values won't be applied to the underlying data, and thus won't be inherited if this object is subsequently used in an operation or data selection command.
```

compared directly. If you don't want objects that share a dimension to
be normalized together in your layout, you can change the ``axiswise``
normalization option to True, making each object be normalized
independently:

.. code:: python

    your_layout.options(axiswise=True)

Similarly, if you have a HoloMap composed of multiple frames in an
animation or controlled with widgets, you can make each frame be
normalized independently by changing ``framewise`` to True:

.. code:: python

    your_holomap.options(framewise=True)

**Q: Why doesn't my DynamicMap respect the framewise=False option for
axis normalization across frames?**

**A:** Unfortunately, HoloViews has no way of knowing the axis ranges
of objects that might be returned by future calls to a DynamicMap's
callback function, and so there is no way for it to fully implement
``framewise=False`` normalization (even though such normalization
is the default in HoloViews). Thus as a special case, a DynamicMap
(whether created specifically or as the return value of various
operations that accept a ``dynamic=True`` argument) will by default
compute its ranges *using the first frame's data only*. If that is not

> the behavior you want, you either set ``framewise=True`` on it to enable

Reviewer: ```suggestion
 the behavior you want, you can either set ``framewise=True`` on it to enable
```

**Q: How do I export a figure?**

**A:** The easiest way to save a figure is the `hv.save` utility,
 which allows saving plots in different formats depending on what is
 supported by the selected backend:

.. code:: python

  # Using bokeh
  hv.save(obj, 'plot.html', backend='bokeh')

  # Using matplotlib
  hv.save(obj, 'plot.svg', backend='matplotlib')

Note that the backend is optional and will default to the currently
activated backend (i.e. ``hv.Store.current_backend``).

**Q: Can I export and customize a bokeh or matplotlib figure directly?**

**A:**: Sometimes it is useful to customize a plot further using the
 underlying plotting API used to render it. The `hv.render` method
 returns the rendered representation of a holoviews object as bokeh or
 matplotlib figure:

.. code:: python

  # Using bokeh
  p = hv.render(obj, backend='bokeh')

  # Using matplotlib
  fig = hv.render(obj, backend='matplotlib')

Note that the backend is optional and will default to the currently
activated backend (i.e. ``hv.Store.current_backend``).

> If instead you want to customize the object before it is plotted it

Reviewer: ```suggestion
If the main reason you want access to the object is to somehow customize it before it is
plotted, instead consider that it
```

is possible to write so called ``hooks``:

.. code:: python

  def hook(plot, element):
    # The bokeh/matplotlib figure
    plot.state

	# A dictionary of handles on plot subobjects, e.g. in matplotlib
	# artist, axis, legend and in bokeh x_range, y_range, glyph, cds etc.
	plot.handles

  hv.Curve(df, 'x_col', 'y_col').options(hooks=[hook])

These hooks can modify the backend specific representation, e.g. the

> matplotlib figure, before allowing arbitrary customizations to be

Reviewer: ```suggestion
matplotlib figure, before it is displayed, allowing arbitrary customizations to be
```
